source/code/plugin/out_mdm.rb

```diff
@@ -167,10 +168,12 @@ def send_to_mdm(post_body)
           # Not raising exception, as that will cause retries to happen
         elsif !response.code.empty? && response.code.start_with?("4")
           # Log 400 errors and continue
+          ApplicationInsightsUtility.sendCustomEvent("AKSCustomMetricsMDMPostError-#{response.code} ", {})
           @log.info "Non-retryable HTTPServerException when POSTing Metrics to MDM #{e} Response: #{response}"
         else
           # raise if the response code is non-400
           @log.info "HTTPServerException when POSTing Metrics to MDM #{e} Response: #{response}"
+          ApplicationInsightsUtility.sendCustomEvent("AKSCustomMetricsMDMPostError-#{response.code} ", {})
```

Comment:
the 'else' part can have response.code as empty.

Reply:
the idea here is this:
we will only get a finite set of response codes. 400s, 500s, or empty. So we should still be able to filter them out. 
The set of event names again is finite, based on the response code.

Reply:
You should use 'dimension' for any metric rather than using metric name for this.  It is easier to do this way but pretty ugly. This will cause metric creep. Please create a back log. We should review all telemetry. Imagine every api call logging metric like this. Its not good & hard to chart & report.